src/ccolamd.jl

```diff
@@ -76,16 +76,15 @@ function ccolamd!(n_row, A::Vector{SuiteSparse_long}, p::Vector{SuiteSparse_long
     return ccolamd!(n_row, A, p, C_NULL, stats, cmember)
 end
 
-function ccolamd!(n_row, A::Vector{SuiteSparse_long}, p::Vector{SuiteSparse_long})
+function ccolamd!(n_row, A::Vector{SuiteSparse_long}, p::Vector{SuiteSparse_long}, constraints=zeros(SuiteSparse_long, length(p) - 1))
     n_col = length(p) - 1
-    cmember = zeros(SuiteSparse_long, length(p) - 1)
-    return ccolamd!(n_row, A, p, cmember)
+    return ccolamd!(n_row, A, p, constraints)
 end
 
-ccolamd(n_row, A::Vector{SuiteSparse_long}, p::Vector{SuiteSparse_long}) =
-    ccolamd!(n_row, copy(A), copy(p))
+ccolamd(n_row, A::Vector{SuiteSparse_long}, p::Vector{SuiteSparse_long}, constraints=zeros(SuiteSparse_long, length(p) - 1)) =
+    ccolamd!(n_row, copy(A), copy(p), constraints)
 
-ccolamd(A::SparseMatrixCSC) = ccolamd(size(A, 1), A.rowval, A.colptr)
+ccolamd(A::SparseMatrixCSC, constraints=zeros(SuiteSparse_long, length(A.colptr) - 1)) = ccolamd(size(A, 1), A.rowval, A.colptr)
```

Comment:
The `constraints` variable, although exposed, is not being used to call the `ccolamd` function. I think it should be:
```jl
... = ccolamd(size(A,1), A.rowval, A.colptr, constraints)
```
with the `constraints` variable at the end.

Reply:
oh good catch thanks -- will update now